### ISSUE TITLE
Drop support for Python 2?

### ISSUE THREAD

**Issue by meowmeowmeowcat** [google/python-fire#420]
Are there any plans for dropping Python 2 support?

**Comment by meowmeowmeowcat** [comment]
If this sounds good, I will put the roadmap here and work on it.

**Comment by hugovk** [comment]
👍 

Python 2 has been EOL for over three years and has very few downloads by now:

| category | percent | downloads |
|:---------|--------:|----------:|
| 3.7      |  34.48% | 2,362,551 |
| 3.8      |  32.81% | 2,248,268 |
| 3.9      |  22.52% | 1,543,519 |
| 3.10     |   5.54% |   379,515 |
| null     |   2.97% |   203,628 |
| 3.6      |   0.75% |    51,450 |
| 3.11     |   0.58% |    39,506 |
| 2.7      |   0.32% |    22,085 |
| 3.5      |   0.03% |     1,991 |
| 3.4      |   0.00% |        81 |
| 3.12     |   0.00% |        71 |
| Total    |         | 6,852,665 |

Date range: 2023-03-01 - 2023-03-31

Source: `pip install -U pypistats && pypistats python_minor fire --last-month`

![image](https://user-images.githubusercontent.com/1324225/231401363-3be10be1-9a01-4d23-aedc-dcdda0e718a3.png)

Source: https://pypistats.org/packages/fire

**Comment by dbieber** [comment]
This upcoming release v0.6.0 will be the last one where we claim support for Python 2.

**Comment by kirk86** [comment]
Sorry maybe this is not the correct place to post, but just wanted to ask if fire supports python 3.10 and 3.11, from the readme seems like it supports only up to 3.9 (if I'm not mistaken) is that correct?

**Comment by dbieber** [comment]
Yes, Python Fire supports these newer versions of Python. We test through 3.12.
v0.6.0 is out now. We're going to drop support for Python 2 moving forward.
